File: Phitron/01. C and CPP/05. Recap module/001. additional/M.lucky_numbers.c
#include <stdio.h>

int main()
{
    int a, b, flag = 0, isEmpty = 0, n, temp, i;
    scanf("%d %d", &a, &b);

    for (i = a; i <= b; i++)
    {
        flag = 0;
        n = i;
        while (n)
        {
            temp = n % 10;
            n = n / 10;
            if (temp != 4 && temp != 7)
            {
                flag = 1;
                break;
            }
        }
        if (flag == 0)
        {
            isEmpty = 1;
            printf("%d ", i);
        }
    }

    if (isEmpty == 0)
    {
        printf("-1");
    }

    return 0;
}

File: Phitron/07. SDT/00. OOP and Python Programming/01. Week 01/Module 2.5/M_Lucky_Numbers.py
a, b = map(int, input().split())
isEmpty = False
flag = False


for i in range(a, b + 1):
    n = i
    flag = False
    while n:
        temp = n % 10
        n = n // 10

        if temp != 4 and temp != 7:
            flag = True
            break

    if not flag:
        isEmpty = True
        print(i, end=" ")

if not isEmpty:
    print("-1")
